### DISCUSSION TITLE
Introduce optional merchant processing fees

### DISCUSSION THREAD

**Discussion by PiotrWajner** [btcpayserver/btcpayserver#1970] [Ideas / Feature Requests]
**Is your feature request related to a problem? Please describe.**

**THE ISSUE** I'm seeing with BTCPayServer is that all people who would like to host a server for other merchants, pay the cost of sucha service and don't get rewarded for this. In other words the host is not being incentivized to participate in the network unless he is a merchant itself.
While BTCPayServer encourages its users to sovereignty by running its own instances, it does not create an environment for better user support. 

**Describe the solution you'd like**
Introduce optional merchant processing fees as of % of transaction. The fees would go to server host/admin. 
The advanced version of this feature could include a referral system where the host owner and referrer split the fees. 

**Describe alternatives you've considered**
I thought about subscription based model and while its a good idea longterm, shortterm it creates a financial barrier hard to get over in early stages of Bitcoin adoption. 
Subscription based model creates a question on merchant side: _'Why should I be paying a subscryption fee if "nobody" is paying in Bitcoin yet?'_

**Additional context**

**JUSTIFICATION:** Bitcoin is about many things, but the two that strike me the most in the context of this feature request is freedom of choice and excellent incentive system where all the participants are incentivized to be part of the payment network. 
BTCPayServer is partially lacking the incentive system. The host benefits only if he is a merchant itself.
**THE BENEFIT** of introducing merchant processing fees would be not only a monetary reward for host owners to cover their costs but also would create an environment where the host owners would actively reach out to other potential merchants not familiar with BTC and spread the love about Bitcoin. Further they would provide all necessary marketing and support (including documentation in many languages etc.) for on-boarding new users (merchants).
At the end of the day, each user has an option to run a self-hosted node and be independent.   

I hope my reasoning show that this feature could be not only beneficial for popularity of BTCPayServer but also Bitcoin itself (and other crypto if you in to that :) )

I personally would be interested in running BTCPayServer and marketing/support campaigns only if the right incentive was in place, and I'm sure many others too.

**Comment by Kukks** [comment]
This requires payment methods to be in some form of custodial form where
the server receives it on behalf of merchants and then redistributes it. I
NACK this concept as a native feature and would only suggest it as a
separate extension module.

**Reply by bolatovumar** [reply]
Agreed.

**Reply by mysticaltech** [reply]
Would be great, honestly, especially for marketplaces!

**Comment by pavlenex** [comment]
Related #204 #1925

**Comment by brainharrington** [comment]
Thanks for keeping track of each of these related conversations @pavlenex, I'd love to help add to this if possible. I am helping run btc pay server for several merchants and have a backlog of others interested. Current pain points I have is how to cash them out on lightning, how to get email notifications turned on, and how to take a commission on their sales. All good issues on #204, #1925, and #1970

**Reply by dennisreimann** [reply]
I want to implement optional service fees in LNbank, because I think it'd be a good fit there. Tracking it here dennisreimann/btcpayserver#7

**Reply by pavlenex** [reply]
I think that's a good idea. Probably the best way to tackle this. Incetivizing LN adoption for local community hosts. Let's go Dennis!

**Comment by c3piio** [comment]
Could it be possible to achieve transaction fees by specifying the xpub of the store, the xpub of the host, and defining a store percentage fee on all transactions to go to the host's wallet? When a customer makes a bitcoin purchase in the online store, btcpayserver creates the invoice and transaction with multiple outputs to the store and host. No custody necessary since the payment goes directly to the store owner's and host's wallets.

**Comment by Kukks** [comment]
You would have to instruct the customer to do the payment with two
destinations unless you are willing to pay the transaction fee to split the
payment in two after receiving from the customer.
